src/Controls/src/BindingSourceGen/BindingSourceGenerator.cs

```diff
@@ -252,31 +252,28 @@ public sealed record CodeWriterBinding(
 
 public sealed record SourceCodeLocation(string FilePath, int Line, int Column);
 
-public sealed record TypeName(string GlobalName, bool IsNullable, bool IsGenericParameter)
+public sealed record TypeName(
+	string GlobalName,
+	bool IsNullable = false,
+	bool IsGenericParameter = false,
+	bool IsValueType = false) // TODO: require setting this explicitly
 {
 	public override string ToString()
 		=> IsNullable
 			? $"{GlobalName}?"
 			: GlobalName;
 }
 
-public sealed record MemberAccess(string MemberName, bool IsNullable) : IPathPart
+public sealed record MemberAccess(string MemberName, bool IsNullable, TypeName? CastTo = null) : IPathPart
```

Comment:
We split `PathPart` into `IndexAccess` and because of the `object? Index` argument. Maybe we should consider refactoring this in the future as well to follow the same pattern?

Reply:
I was considering something like `record Cast(IPathPart Part, TypeName TargetType) : IPathPart`. It would definitely be cleaner but I went for the simpler dirty solution. I'll sleep on it and revisit it tomorrow morning.

Reply:
You know what, there's no need to wait until tomorrow :D

Reply:
Just an idea for future: maybe we could go full on decorator on that. Like having `MemberAccess` and then `WithIndexer` and `WithCast` wrappers?